File: debug_job.js

const admin = require('firebase-admin');
const { getFirestore } = require('firebase-admin/firestore');

// Initialize with application default credentials (works if GOOGLE_APPLICATION_CREDENTIALS is set)
// Or reuse existing config logic.
// Since we are in the project root, we can try to use the project's firebase_admin lib if compiling TS, 
// but a standalone JS script is safer/faster for debug.

// We need the service account. It's in .env.local usually.
require('dotenv').config({ path: '.env.local' });

const serviceAccount = {
    projectId: process.env.FIREBASE_PROJECT_ID,
    clientEmail: process.env.FIREBASE_CLIENT_EMAIL,
    privateKey: (process.env.FIREBASE_PRIVATE_KEY || "").replace(/\\n/g, '\n'),
};

if (!admin.apps.length) {
    admin.initializeApp({
        credential: admin.credential.cert(serviceAccount),
    });
}

const db = getFirestore();

async function checkJob(jobId) {
    console.log(`Checking Job: ${jobId}`);
    const doc = await db.collection('jobs').doc(jobId).get();
    if (!doc.exists) {
        console.log("Job not found!");
    } else {
        console.log("Job Data:", JSON.stringify(doc.data(), null, 2));
    }
}

const jobId = process.argv[2];
if (jobId) {
    checkJob(jobId).catch(console.error);
} else {
    console.log("Please provide Job ID");
}

File: debug_job.py
import firebase_admin
from firebase_admin import credentials, firestore
import os
import json
from datetime import datetime, timedelta

# Initialize Firebase (Local Script)
if not firebase_admin._apps:
    cred = credentials.Certificate('serviceAccountKey.json')
    firebase_admin.initialize_app(cred)

db = firestore.client()

def check_recent_job_and_leads():
    print("--- 1. Checking Recent Jobs ---")
    jobs_ref = db.collection('jobs').order_by('createdAt', direction=firestore.Query.DESCENDING).limit(1)
    jobs = list(jobs_ref.stream())
    
    if not jobs:
        print("No jobs found.")
        return

    job = jobs[0]
    job_data = job.to_dict()
    job_id = job.id
    print(f"Latest Job ID: {job_id}")
    print(f"Status: {job_data.get('status')}")
    print(f"Query: {job_data.get('query')}")
    print(f"Logs: {len(job_data.get('logs', []))} entries")
    
    if job_data.get('logs'):
        print("Last Leg: ", job_data['logs'][-1])

    print("\n--- 2. Checking Leads for Job ---")
    leads_ref = db.collection('leads').where('jobId', '==', job_id).limit(5)
    leads = list(leads_ref.stream())
    
    print(f"Leads Found: {len(leads)}")
    for lead in leads:
        data = lead.to_dict()
        print(f"- {data.get('BusinessName', 'NoName')} ({data.get('City')})")

if __name__ == "__main__":
    check_recent_job_and_leads()
